src/main/java/de/tum/in/www1/artemis/service/connectors/BambooService.java

```diff
@@ -717,18 +731,12 @@ public Result addFeedbackToResultNew(Result result, List<Object> jobs) {
                     String methodName = (String) failedTest.get("name"); // in the attribute "methodName", bamboo seems to apply some unwanted logic
 
                     List<String> errors = (List<String>) failedTest.get("errors");
-                    String errorMessageString = "";
-                    for (String error : errors) {
-                        //Splitting string at the first linebreak to only get the first line of the Exception
-                        errorMessageString += error.split("\\n", 2)[0] + "\n";
-                    }
-
-                    //TODO: if PE.Language == C, do not split, else do split (take only first line)
-                    //TODO: filter java.lang.AssertionError
+                    final StringBuilder errorMessageString = new StringBuilder();
+                    errors.forEach(error -> errorMessageString.append(processResultErrorMessage(programmingLanguage, error)));
```

Comment:
I think we can also apply @thilo-behnke's comment from above here:
```java
errors.stream()
  .map(errorString -> processResultErrorMessage(programmingLanguage, errorString) + "\\n")
  .reduce("", String::concat);
```

Reply:
True!